reference/enterprise-api-ref/health-diagnostic.markdown

```diff
@@ -0,0 +1,135 @@
+* **sortColumn** *(string)*
+    Column name on which to sort results. Optional parameter.
+* **sortDescending** *(boolean)*
+    Sorting order. Optional parameter.
+* **skip** *(integer)*
+    Number of results to skip for the processed
+    query. The Mission Portal uses this for pagination. Optional parameter.
+* **limit**  *(integer)*
+    Limit the number of results in the query.
+* **hostContextInclude** *(array)*
+    Includes only results that concern hosts which have all specified CFEngine contexts (class) set. Optional parameter.
+* **hostContextExclude** *(array)*
+    Excludes results that concern hosts which have specified CFEngine context (class) set. Hosts that have at lest one of the specified contexts set will be excluded from the results. Optional parameter.
+
+**CURL Request Example:**
+```
+curl -k --user <username>:<password> -X POST \
+  https://hub.cfengine.com/api/health-diagnostic/report/agentNotRunRecently \
+  -H 'Content-Type: application/json' \
+  -d '{"limit": 50}'
+```
+
+**Example response:**
+
+```
+{
+    "data": [
+        {
+            "header": [
+                {
+                    "columnName": "key",
+                    "columnType": "STRING"
+                },
+                {
+                    "columnName": "Host name",
+                    "columnType": "STRING"
+                },
+                {
+                    "columnName": "Last report collected",
+                    "columnType": "STRING"
+                },
+                {
+                    "columnName": "Last agent local execution time",
+                    "columnType": "STRING"
+                }
+            ],
+            "query": "SELECT h.Hostkey as key,h.hostname as \"Host name\", lastreporttimestamp as \"Last report collected\", agentstatus.lastagentlocalexecutiontimestamp as \"Last agent local execution time\"  \n                            FROM vm_hosts h \n                            LEFT JOIN agentstatus  ON agentstatus.Hostkey = h.Hostkey WHERE h.HostKey IN (SELECT result.hostkey FROM (SELECT agentstatus.HostKey \n                      FROM agentstatus \n                      LEFT JOIN vm_hosts ON vm_hosts.hostkey = agentstatus.hostkey\n                      WHERE  extract(epoch from (lastReportTimeStamp::timestamp - lastagentlocalexecutiontimestamp::timestamp)) > agentexecutioninterval::int  * 1.3) AS result \n                      WHERE hostkey IS NOT NULL  AND HostKey NOT IN (SELECT hostkey FROM hosts_not_reported) AND HostKey NOT IN (SELECT Hosts_view.HostKey \n                      FROM vm_hosts Hosts_view \n                      WHERE Hosts_view.lastreporttimestamp < to_timestamp('1549559891')) AND HostKey NOT IN (SELECT SameHosts.HostKey \n                      FROM (\n                          SELECT remotehostkey as HostKey FROM lastseenhostslogs GROUP BY remotehostkey HAVING COUNT(distinct remotehostip) > 1\n                          ) AS SameHosts))",
```

Comment:
Interesting. It returns the sql used to get it? Can this sql be used directly in the custom reporting ui?

Reply:
Yes, it returns the SQL as query API does. And yes this SQL can be used directly in the custom reporting UI.